src/twisted/internet/defer.py

```diff
@@ -1557,7 +1599,8 @@ class _InternalInlineCallbacksCancelledError(Exception):
     """
 
 
-
+@deprecated(Version('Twisted', 20, 4, 0),
```

Comment:
The correct version for new deprecations is `Version('Twisted', 'NEXT', 0, 0)`.

Reply:
@mthuurne I get this:

```
Traceback (most recent call last):
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/web/_flatten.py", line 315, in _flattenTree
    element = next(stack[-1])
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/web/_flatten.py", line 231, in _flattenElement
    result = renderMethod(request, rootClone)
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/pydoctor/templatewriter/summary.py", line 300, in stuff
    undoccedpublic = [o for o in self.system.allobjects.values()
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/pydoctor/templatewriter/summary.py", line 301, in <listcomp>
    if o.isVisible and not hasdocstring(o)]
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/pydoctor/templatewriter/summary.py", line 271, in hasdocstring
    for source in ob.docsources():
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctor.py", line 158, in docsources
    getDeprecated(self, list(self.decorators))
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctor.py", line 90, in getDeprecated
    self, self.name, a)
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctor.py", line 140, in deprecatedToUsefulText
    version = versionToUsefulObject(deprecated.args[0])
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctor.py", line 130, in versionToUsefulObject
    return Version(version.args[0].s, *[x.n for x in version.args[1:] if x])
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctor.py", line 130, in <listcomp>
    return Version(version.args[0].s, *[x.n for x in version.args[1:] if x])
AttributeError: 'Str' object has no attribute 'n'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/circleci/documentation/bin/admin/build-apidocs", line 12, in <module>
    BuildAPIDocsScript().main(sys.argv[1:])
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_release.py", line 498, in main
    self.buildAPIDocs(FilePath(args[0]), FilePath(args[1]))
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_release.py", line 483, in buildAPIDocs
    output)
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_release.py", line 341, in build
    main(args)
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/pydoctor/driver.py", line 348, in main
    writer.writeModuleIndex(system)
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/pydoctor/templatewriter/writer.py", line 56, in writeModuleIndex
    flattenToFile(f, page)
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/pydoctor/templatewriter/writer.py", line 21, in flattenToFile
    raise err[0]
twisted.web.error.FlattenerError: Exception while flattening:
  <pydoctor.templatewriter.summary.UndocumentedSummaryPage object at 0x7f21c6af3a20>
  [Tag('html', children=['\n', '  ', Tag('head', children=['\n', '    ', Tag('title', children=['Something']), '\n', '    ', Tag('meta', attributes=OrderedDict([('http-equiv', 'Content-Type'), ('content', 'text/html;charset=utf-8')])), '\n', '    ', Tag('link', attributes=OrderedDict([('rel', 'stylesheet'), ('type', 'text/css'), ('href', 'bootstrap.min.css')])), '\n', '    ', Tag('link', attributes=OrderedDict([('rel', 'stylesheet'), ('type', 'text/css'), ('href', 'apidocs.css')])), '\n', '  ']), '\n', '  ', Tag('body', children=['\n', '    ', Tag('nav', attributes=OrderedDict([('class', 'navbar navbar-default')]), children=['\n', '      ', Tag('div', attributes=OrderedDict([('class', 'container')]), children=['\n', '\n', '        ', Tag('div', attributes=OrderedDict([('class', 'navbar-header')]), children=['\n', '          ', Tag('a', attributes=OrderedDict([('class', 'navbar-brand'), ('href', 'index.html')]), children=['\n', '            ', Tag('', children=['Some Project']), ' API Documentation', '\n', '          ']), '\n', '        ']), '\n', '      ']), '\n', '    ']), '\n', '\n', '    ', Tag('div', attributes=OrderedDict([('style', 'display: none'), ('id', 'current-docs-container'), ('class', 'container')]), children=['\n', '      ', Tag('div', attributes=OrderedDict([('class', 'col-sm-12')]), children=['\n', '        ', Tag('a', attributes=OrderedDict([('id', 'current-docs-link')]), children=['\n', '          Go to the latest version of this document.', '\n', '        ']), '\n', '      ']), '\n', '    ']), '\n', '\n', '    ', Tag('div', attributes=OrderedDict([('class', 'container')]), children=['\n', '\n', '      ', Tag('div', attributes=OrderedDict([('class', 'page-header')]), children=['\n', '        ', Tag('h1', children=['Heading']), '\n', '      ']), '\n', '\n', '      ', Tag('ul', children=['\n', '      ']), '\n', '\n', '    ']), '\n', '\n', '    ', Comment(' Google analytics, obviously. '), '\n', '    ', Tag('script', attributes=OrderedDict([('src', '//www.google-analytics.com/urchin.js'), ('type', 'text/javascript')])), '\n', '    ', Tag('script', attributes=OrderedDict([('type', 'text/javascript')]), children=['\n', '      _uacct = "UA-99018-6";', '\n', '      urchinTracker();', '\n', '    ']), '\n', '\n', '    ', Comment(" If the documentation isn't current, insert a current link. "), '\n', '    ', Tag('script', attributes=OrderedDict([('type', 'text/javascript')]), children=['\n', "      if (window.location.pathname.indexOf('/current/') == -1) {", '\n', '        ', Comment(' Give the user a link to this page, but in the current version of the docs. '), '\n', "        var link = document.getElementById('current-docs-link');", '\n', "        link.href = window.location.pathname.replace(/\\/\\d+\\.\\d+\\.\\d+\\/api\\//, '/current/api/');", '\n', '        ', Comment(' And make it visible '), '\n', "        var container = document.getElementById('current-docs-container');", '\n', '        container.style.display = "";', '\n', '        delete link;', '\n', '        delete container;', '\n', '      }', '\n', '    ']), '\n', '\n', '  ']), '\n'])]
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctortemplates/summary.html", line 2, column 0, in "html"
  ['\n', '  ', Tag('head', children=['\n', '    ', Tag('title', children=['Something']), '\n', '    ', Tag('meta', attributes=OrderedDict([('http-equiv', 'Content-Type'), ('content', 'text/html;charset=utf-8')])), '\n', '    ', Tag('link', attributes=OrderedDict([('rel', 'stylesheet'), ('type', 'text/css'), ('href', 'bootstrap.min.css')])), '\n', '    ', Tag('link', attributes=OrderedDict([('rel', 'stylesheet'), ('type', 'text/css'), ('href', 'apidocs.css')])), '\n', '  ']), '\n', '  ', Tag('body', children=['\n', '    ', Tag('nav', attributes=OrderedDict([('class', 'navbar navbar-default')]), children=['\n', '      ', Tag('div', attributes=OrderedDict([('class', 'container')]), children=['\n', '\n', '        ', Tag('div', attributes=OrderedDict([('class', 'navbar-header')]), children=['\n', '          ', Tag('a', attributes=OrderedDict([('class', 'navbar-brand'), ('href', 'index.html')]), children=['\n', '            ', Tag('', children=['Some Project']), ' API Documentation', '\n', '          ']), '\n', '        ']), '\n', '      ']), '\n', '    ']), '\n', '\n', '    ', Tag('div', attributes=OrderedDict([('style', 'display: none'), ('id', 'current-docs-container'), ('class', 'container')]), children=['\n', '      ', Tag('div', attributes=OrderedDict([('class', 'col-sm-12')]), children=['\n', '        ', Tag('a', attributes=OrderedDict([('id', 'current-docs-link')]), children=['\n', '          Go to the latest version of this document.', '\n', '        ']), '\n', '      ']), '\n', '    ']), '\n', '\n', '    ', Tag('div', attributes=OrderedDict([('class', 'container')]), children=['\n', '\n', '      ', Tag('div', attributes=OrderedDict([('class', 'page-header')]), children=['\n', '        ', Tag('h1', children=['Heading']), '\n', '      ']), '\n', '\n', '      ', Tag('ul', children=['\n', '      ']), '\n', '\n', '    ']), '\n', '\n', '    ', Comment(' Google analytics, obviously. '), '\n', '    ', Tag('script', attributes=OrderedDict([('src', '//www.google-analytics.com/urchin.js'), ('type', 'text/javascript')])), '\n', '    ', Tag('script', attributes=OrderedDict([('type', 'text/javascript')]), children=['\n', '      _uacct = "UA-99018-6";', '\n', '      urchinTracker();', '\n', '    ']), '\n', '\n', '    ', Comment(" If the documentation isn't current, insert a current link. "), '\n', '    ', Tag('script', attributes=OrderedDict([('type', 'text/javascript')]), children=['\n', "      if (window.location.pathname.indexOf('/current/') == -1) {", '\n', '        ', Comment(' Give the user a link to this page, but in the current version of the docs. '), '\n', "        var link = document.getElementById('current-docs-link');", '\n', "        link.href = window.location.pathname.replace(/\\/\\d+\\.\\d+\\.\\d+\\/api\\//, '/current/api/');", '\n', '        ', Comment(' And make it visible '), '\n', "        var container = document.getElementById('current-docs-container');", '\n', '        container.style.display = "";', '\n', '        delete link;', '\n', '        delete container;', '\n', '      }', '\n', '    ']), '\n', '\n', '  ']), '\n']
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctortemplates/summary.html", line 9, column 2, in "body"
  ['\n', '    ', Tag('nav', attributes=OrderedDict([('class', 'navbar navbar-default')]), children=['\n', '      ', Tag('div', attributes=OrderedDict([('class', 'container')]), children=['\n', '\n', '        ', Tag('div', attributes=OrderedDict([('class', 'navbar-header')]), children=['\n', '          ', Tag('a', attributes=OrderedDict([('class', 'navbar-brand'), ('href', 'index.html')]), children=['\n', '            ', Tag('', children=['Some Project']), ' API Documentation', '\n', '          ']), '\n', '        ']), '\n', '      ']), '\n', '    ']), '\n', '\n', '    ', Tag('div', attributes=OrderedDict([('style', 'display: none'), ('id', 'current-docs-container'), ('class', 'container')]), children=['\n', '      ', Tag('div', attributes=OrderedDict([('class', 'col-sm-12')]), children=['\n', '        ', Tag('a', attributes=OrderedDict([('id', 'current-docs-link')]), children=['\n', '          Go to the latest version of this document.', '\n', '        ']), '\n', '      ']), '\n', '    ']), '\n', '\n', '    ', Tag('div', attributes=OrderedDict([('class', 'container')]), children=['\n', '\n', '      ', Tag('div', attributes=OrderedDict([('class', 'page-header')]), children=['\n', '        ', Tag('h1', children=['Heading']), '\n', '      ']), '\n', '\n', '      ', Tag('ul', children=['\n', '      ']), '\n', '\n', '    ']), '\n', '\n', '    ', Comment(' Google analytics, obviously. '), '\n', '    ', Tag('script', attributes=OrderedDict([('src', '//www.google-analytics.com/urchin.js'), ('type', 'text/javascript')])), '\n', '    ', Tag('script', attributes=OrderedDict([('type', 'text/javascript')]), children=['\n', '      _uacct = "UA-99018-6";', '\n', '      urchinTracker();', '\n', '    ']), '\n', '\n', '    ', Comment(" If the documentation isn't current, insert a current link. "), '\n', '    ', Tag('script', attributes=OrderedDict([('type', 'text/javascript')]), children=['\n', "      if (window.location.pathname.indexOf('/current/') == -1) {", '\n', '        ', Comment(' Give the user a link to this page, but in the current version of the docs. '), '\n', "        var link = document.getElementById('current-docs-link');", '\n', "        link.href = window.location.pathname.replace(/\\/\\d+\\.\\d+\\.\\d+\\/api\\//, '/current/api/');", '\n', '        ', Comment(' And make it visible '), '\n', "        var container = document.getElementById('current-docs-container');", '\n', '        container.style.display = "";', '\n', '        delete link;', '\n', '        delete container;', '\n', '      }', '\n', '    ']), '\n', '\n', '  ']
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctortemplates/summary.html", line 29, column 4, in "div"
  ['\n', '\n', '      ', Tag('div', attributes=OrderedDict([('class', 'page-header')]), children=['\n', '        ', Tag('h1', children=['Heading']), '\n', '      ']), '\n', '\n', '      ', Tag('ul', children=['\n', '      ']), '\n', '\n', '    ']
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctortemplates/summary.html", line 35, column 6, in "ul"
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/web/_flatten.py", line 315, in _flattenTree
    element = next(stack[-1])
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/web/_flatten.py", line 231, in _flattenElement
    result = renderMethod(request, rootClone)
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/pydoctor/templatewriter/summary.py", line 300, in stuff
    undoccedpublic = [o for o in self.system.allobjects.values()
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/pydoctor/templatewriter/summary.py", line 301, in <listcomp>
    if o.isVisible and not hasdocstring(o)]
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/pydoctor/templatewriter/summary.py", line 271, in hasdocstring
    for source in ob.docsources():
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctor.py", line 158, in docsources
    getDeprecated(self, list(self.decorators))
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctor.py", line 90, in getDeprecated
    self, self.name, a)
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctor.py", line 140, in deprecatedToUsefulText
    version = versionToUsefulObject(deprecated.args[0])
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctor.py", line 130, in versionToUsefulObject
    return Version(version.args[0].s, *[x.n for x in version.args[1:] if x])
  File "/home/circleci/documentation/build/apidocs/lib/python3.6/site-packages/twisted/python/_pydoctor.py", line 130, in <listcomp>
    return Version(version.args[0].s, *[x.n for x in version.args[1:] if x])
AttributeError: 'Str' object has no attribute 'n'
```

Reply:
I fixed the parser and now it works